packages/data-schema/src/ModelType.ts

```diff
@@ -238,6 +251,16 @@ export type ModelType<
       SetTypeSubArg<T, 'identifier', PrimaryIndexIR>,
       UsedMethod | 'identifier'
     >;
+
+    /**
+     * Adds secondary index for a model, secondary index consists of a "hash key" and optionally, a "sort key"
+     *
+     * @param callback A function that specifies "hash key" and "sort key" 
+     * @returns A ModelType instance with updated secondary index
+     *
+     * @example
+     * model().secondaryIndexes((index) => [index('type').sortKeys(['sort'])])
```

Comment:
```suggestion
     * model().secondaryIndexes((index) => [index('type').sortKeys(['sort'])])
     *
     * @see [Amplify documentation for secondary indexes](https://docs.amplify.aws/react/build-a-backend/data/data-modeling/secondary-index/)
     * @see [Amazon DynamoDB documentation for secondary indexes](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/SecondaryIndexes.html)
```

it'd be helpful to add a few documentation links to these snippets, especially for concepts the customer may not be exposed to ahead of time